_posts/2023-11-07-sssd-authentication-method-policy.md

```diff
@@ -0,0 +1,48 @@
+---
+layout:     post
+title:      "SSSD: authentication method policy"
+date:       2023-11-07 12:00:00 +0100
+categories: idm
+---
+
+# Introduction
+
+SSSD has recently released version [2.9.2](https://sssd.io/release-notes/sssd-2.9.2.html), which adds a new option to control the authentication method policy. The need for this new option comes from adding new authentication methods such as passkey[1][2]. It is possible that more authentication methods will emerge in the future, so it seems necessary to add a way to control the policy governing them.
+
+# Existing policy before v2.9.2
+
+When several authentication mechanisms were available SSSD defaulted to try methods that would return a Kerberos ticket. If the server was unavailable, then SSSD would try an offline authentication. For this purpose, it would match the preferred online authentication method, i.e. if the online authentication would be performed via password, the offline authentication would be the same.
+

```

Comment:
This sounds a bit like, if you used the password for online authentication you have to use ti for offline authentication as well even if other methods like e.g. Smartcard authentication would have been possible as well?

I would try to indicate that whatever methods are available online, the same will be avaialble offline as well with the restriction that a) for password authentication `cache_credentials = True` must be set and there must be a successful online authentication recorded in the cache. And b) for Smartcard authentication there has to be a successful online authentication as well to stored the proper mapping in the cache. Afaik passkey should work already after the user is looked up once and all requred attributes are stored in the cache and external IdP only works online.